ts/screens/authentication/IdpLoginScreen.tsx

```diff
@@ -96,32 +143,71 @@ const onNavigationStateChange = (
 class IdpLoginScreen extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
-
     this.state = {
       requestState: pot.noneLoading
     };
   }
 
-  private handleOnError = (): void =>
+  private traceInOut(boundaryState: AuthState) {
+    trackAuth({
+      state: boundaryState,
+      detail: this.props.loggedOutWithIdpAuth
+        ? this.props.loggedOutWithIdpAuth.idp.entityID
+        : "idp n/a"
+    });
+  }
+
+  public componentWillUnmount() {
+    this.traceInOut("IDPL_START");
+  }
+
+  public componentDidMount() {
+    this.traceInOut("IDPL_END");
+  }
+
+  private handleOnError = (): void => {
+    trackAuth({ state: "IDPL_ERROR", detail: "request network error" });
     this.setState({
       requestState: pot.noneError("error")
     });
+  };
 
   private goBack = this.props.navigation.goBack;
 
   private setRequestStateToLoading = () =>
     this.setState({ requestState: pot.noneLoading });
 
   private handleNavigationStateChange = (event: NavState): void => {
+    if (
+      pot.isSome(this.state.requestState) &&
+      event.url &&
+      event.url !== this.state.requestState.value
+    ) {
+      trackAuth({ state: "IDPL_URL_CHANGED", detail: event.url });
```

Comment:
Url may contain sensitive information we don't want (or are not allowed) to log.

Reply:
@fpersico makes a good point! cc @matteodesanti 

sensitive tokens can be embedded in the URL (also in the path), we can't implement a general rule to remove them

Reply:
@cloudify The url contains the whole saml request :)